vhdl_lang/src/ast/search.rs

```diff
@@ -38,6 +38,31 @@ impl SearchState {
     }
 }
 
+#[derive(PartialEq, Debug, Clone)]
+pub enum FoundDeclaration<'a> {
```

Comment:
I think this type is redundant with `NamedEntity` in `named_entity.rs` The goal of analysis is to build a data structure of declarative regions with symbols (Which the LRM calls named entites). The idea was that `Reference` type should point to such a `NamedEntity` instead of just a `SrcPos` like it does today. The `SrcPos` was only a shortcut to get the goto-definiton feature which did not need to know anything more about the named entity other than the source position.